feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/domain/referendum/vote/validations/VoteReferendaValidationPayload.kt

```diff
@@ -8,12 +8,12 @@ import io.novafoundation.nova.feature_wallet_api.presentation.model.DecimalFee
 import io.novafoundation.nova.runtime.multiNetwork.runtime.types.custom.vote.Conviction
 import java.math.BigDecimal
 
-data class VoteReferendumValidationPayload(
-    val onChainReferendum: OnChainReferendum,
+data class VoteReferendaValidationPayload(
+    val onChainReferenda: List<OnChainReferendum>,
     val asset: Asset,
-    val trackVoting: Voting?,
-    val voteAmount: BigDecimal,
-    val fee: DecimalFee,
+    val trackVoting: List<Voting>,
+    val maxAmount: BigDecimal,
     val voteType: VoteType?,
-    val conviction: Conviction
+    val conviction: Conviction?,
```

Comment:
Those nullable fields are quite hard to understand (why is nulalble, when is it nullable) and we will certainly face issues with them in the future when modifing this code
It is better to use a sealed class to describe possible cases